llvm/include/llvm/IR/RepoTicket.h

```diff
@@ -59,46 +58,55 @@ struct GONumber {
 
 /// A structure of a global object (GO) information.
 struct GOInfo {
-  /// GO's initial hash value which does not include the hash of its dependents.
+  /// The Initial-digest is the hash of an object (a function or global
+  /// variable) without consideration for any references to other objects.
   DigestType InitialDigest;
-  /// A set of global objects which the GO's hash depenendent on.
-  DependenciesType Dependencies;
-  /// A set of global variables which the GO's hash contributed to.
-  ContributionsType Contributions;
+  /// The Contributions of an object X are those objects Y which transitively
+  /// reference X and where a potential optimisation to either X or any of Y may
+  /// invalidate both.
+  SmallGOVec Contributions;
+  /// The Dependencies of an object X are those objects Y which X transitively
+  /// references and which are not Contributions.
+  SmallGOVec Dependencies;
 
   GOInfo() = default;
-  GOInfo(DigestType &&Digest, DependenciesType &&Dependencies,
-         ContributionsType &&Contributions)
-      : InitialDigest(std::move(Digest)), Dependencies(std::move(Dependencies)),
-        Contributions(std::move(Contributions)) {}
+  GOInfo(DigestType &&Digest, SmallGOVec &&Contributions,
+         SmallGOVec &&Dependencies)
+      : InitialDigest(std::move(Digest)),
+        Contributions(std::move(Contributions)),
+        Dependencies(std::move(Dependencies)) {}
 };
 using GOInfoMap = DenseMap<const GlobalObject *, GOInfo>;
 
-/// A map from a global variable (GV) to the contibutions.
-using GVInfoMap = DenseMap<const GlobalVariable *, DependenciesType>;
-
-/// A tuple containing the global object information and two unsigned values
-/// which are the number of global variables and functions respectively.
-using ModuleTuple = std::tuple<GOInfoMap, unsigned, unsigned>;
-
-/// Calculate the initial hash value and dependent lists for the global object
-/// 'G'. The value does not include the hash value of its dependent globals.
+/// Calculate the initial hash value, dependencies and contributions for the
+/// global object 'G'. The value does not include the hash value of its
```

Comment:
The second sentence repeats from the definition above, line 61, so I think is unnecessary.

Reply:
Removed, thanks.